### DISCUSSION TITLE
ArrayField fields as textareas instead of inputs

### DISCUSSION THREAD

**Discussion by hertilizer** [bhch/django-jsonform#177] [Q&A] [answered]
<img width="504" alt="Screenshot 2024-09-10 at 1 58 42 PM" src="https://github.com/user-attachments/assets/06b13d0b-26b7-4992-a98b-47df2c189f2b">

I temporarily updated the react-json-form.js to use textarea instead of input for each of the array element fields. This temporary fix will get overridden each time I pip install though. Is there already a built in way of accomplishing this?

**Comment by bhch** [comment, marked as answer]
Use the `widget` keyword:

```diff
{
   "type": "array",
   "items": {
     "type": "string",
+    "widget": "textarea"
   }
}
```

**Reply by hertilizer** [reply]
What is the context of this code? Where does it go?

**Reply by hertilizer** [reply]
This looks like it would be a schema for a JSONField. what about if I am trying to use the ArrayField though?

**Reply by bhch** [reply]
Oops! Sorry. Misread your question.

I'll release a new version which allows you to supply your own schema to the `ArrayField`. It's been a frequently requested feature. I'll try to get it out in a couple of days.

**Reply by bhch** [reply]
@hertilizer Released in [v2.23.0](https://github.com/bhch/django-jsonform/releases/tag/v2.23.0).
